### DISCUSSION TITLE
"Smart resize" functionality?

### DISCUSSION THREAD

**Discussion by ghost** [qtile/qtile#2739] [Q&A] [answered]
Is there a way to get any kind of "smart resize" keybindings like [this](https://github.com/2ver/dotfiles/blob/main/.bin/i3resize) or [this](https://github.com/2ver/dotfiles/blob/main/.config/sxhkd/sxhkdrc#L218-L230) (basically, the ability to grow and shrink windows with the same keybinding based on context) in qTile? Right now, if I have two windows open side by side in the qTile Bsp layout and I want to make the right one smaller, I must first focus the left one and then make it bigger. I'd like the same keybinding that would make the left one larger (if focused) also make right one smaller if it the one I'm currently focused on. Any ideas? Thanks.

**Comment by m-col** [comment, marked as answer]
This is up to the `Layout` itself to define, e.g. the `Columns` layout can do this. If it is obvious based on context what is intended then it could be implemented in the `Bsp` layout too.

**Reply by ghost** [reply]
Ah I see. So basically someone with more experience in python/qTile than myself would need to add that functionality in. Maybe I'll explore some of the other layouts for now because I don't think I can live without that feature :sweat_smile:.

**Reply by m-col** [reply]
If you're familiar with the behaviour of the layout then it might be easy to understand how the code works - and then it might not be too difficult implementing it :)
